examples/vault-cluster-authentication-iam/startup-script-vault.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+# This script is meant to be run as the Startup Script of each Compute Instance
+# while it's booting. The script uses the run-consul and run-vault scripts to
+# configure and start both Vault and Consul in client mode, and then, after initializing
+# and unsealing vault, it configures vault authentication and writes an example
+# that can be read by a client. This script assumes it's running in a Compute Instance
+# based on a Google Image built from the Packer template in examples/vault-consul-image/vault-consul.json.
+#
+# For more information about GCP auth, please refer to https://www.vaultproject.io/docs/auth/gcp.html
+# ==========================================================================
+
+set -e
+
+# Send the log output from this script to startup-script.log, syslog, and the console
+# Inspired by https://alestic.com/2010/12/ec2-user-data-output/
+exec > >(tee /var/log/startup-script.log|logger -t startup-script -s 2>/dev/console) 2>&1
+
+# The Packer template puts the TLS certs in these file paths
+readonly VAULT_TLS_CERT_FILE="/opt/vault/tls/vault.crt.pem"
+readonly VAULT_TLS_KEY_FILE="/opt/vault/tls/vault.key.pem"
+
+# Note that any variables below with <dollar-sign><curly-brace><var-name><curly-brace> are expected to be interpolated by Terraform.
+/opt/consul/bin/run-consul --client --cluster-tag-name "${consul_cluster_tag_name}"
+/opt/vault/bin/run-vault --gcs-bucket ${vault_cluster_tag_name} --tls-cert-file "$VAULT_TLS_CERT_FILE" --tls-key-file "$VAULT_TLS_KEY_FILE" ${enable_vault_ui}
+
+# Log the given message. All logs are written to stderr with a timestamp.
+function log {
+ local -r message="$1"
+ local readonly timestamp=$(date +"%Y-%m-%d %H:%M:%S")
+ >&2 echo -e "$timestamp $message"
+}
+
+# A retry function that attempts to run a command a number of times and returns the output
+function retry {
```

Comment:
TODO for our backlog: move all the Vault/Consul repos to [bash-commons](https://github.com/gruntwork-io/bash-commons) so we can stop copy/pasting this same code again and again. And I suppose you could add your `retry` function to `bash-commons` too!

Reply:
I was actually thinking about this too while updating the bash commons repo with the gruntwork badge!!